### ISSUE TITLE
[🐛 Bug]: Webdriver Jasmine expect(element).toBePresent() doesnt work with latest Appium 2.0 beta48 build

### ISSUE THREAD

**Issue by gforepsly** [webdriverio/webdriverio#9574]
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md#reporting-new-issues).

### WebdriverIO Version

7.29.1

### Node.js Version

14.21.2

### Mode

Standalone Mode

### Which capabilities are you using?

```typescript
{
        platformName: 'iOS',
        "appium:iosInstallPause": 5000,
        "appium:deviceName": process.env.DEVICE_MODEL || 'iPhone 14 Pro',
        "appium:language": 'en',
        // udid: 'auto', // can be used for real devices
        "appium:platformVersion": process.env.IOS_VERSION || '16.2',
        "appium:automationName": 'XCUITest',
        "appium:app": process.env.IOS_BUILD_APP_PATH || join(process.cwd(), './appium/apps/Repsly.app'),
        "appium:newCommandTimeout": 10000,
        "appium:wdaLaunchTimeout": 300000,
        "appium:wdaConnectionTimeout": 600000,
        "appium:wdaStartupRetries": 4,
        "appium:wdaStartupRetryInterval": 120000,
        "appium:wdaLocalPort": process.env.WDA_PORT || 8100
    }

Note:
 - Running latest webdriverIO version in sync mode, using node 15 or any version above doesn't work at all - the issue is when it has to interact with elemnts over appium server, it just fails even tough it connects properly, so we have to downgrade to latest Node 14 and some higher NPM to start the appium 2.0+
```

### What happened?

Writing close like this inside IT block doesn't work (the test is running normally, but when it's time to check for presence, it fails on that step, saying the element is not present - the usual fail):
`expect($('~Workday history')).toBePresent()`

Error Example (a real mess):
`Error: Expected Element({ sessionId: '5ca4d36b-e54d-41aa-a6bd-d32742c7a490', elementId: '0C000000-0000-0000-CC1E-000000000000', element-6066-11e4-a52e-4f735466cecf: '0C000000-0000-0000-CC1E-000000000000', selector: '~Get started', parent: Browser({ sessionId: '5ca4d36b-e54d-41aa-a6bd-d32742c7a490', capabilities: Object({ webStorageEnabled: false, locationContextEnabled: false, browserName: '', platform: 'MAC', javascriptEnabled: true, databaseEnabled: false, takesScreenshot: true, networkConnectionEnabled: false, platformName: 'iOS', iosInstallPause: 5000, deviceName: 'iPhone 14 Pro', language: 'en', platformVersion: '16.2', automationName: 'XCUITest', app: '/Users/gfor/Documents/AutomationProject/TestAutomation/appium/apps/Repsly.app', newCommandTimeout: 10000, wdaLaunchTimeout: 300000, wdaConnectionTimeout: 600000, wdaStartupRetries: 4, wdaStartupRetryInterval: 120000, wdaLocalPort: 8668, udid: '96E22F2B-5DA1-4C51-973F-08E38416AA22' }), addCommand: Function, overwriteCommand: Function, addLocato ... to be present.`

Writing close like this inside IT block works (no fails, all good):
`expect($('~Workday history').isExisting()).toBe(true)`

Error log when this fails (good logs):
`Error: Element isn't visible after 15000 seconds. Error: Error: element ("~Allow Once") still not displayed after 15000ms`

We are using webdriverIO in sync mode, it's a typescript project, appium server 2.0.0-beta.48 under MAC 13.1 Ventura. I will provide more configuration stuff below. 

Can anyone tell me what are we doing wrong? Why are this check above failing, but passing when we change the syntax a bit? We are still on older Node and older sync packages.

### What is your expected behavior?

`expect($('~Workday history')).toBePresent()` 
The above line should work, alongside with the other methods for toBeDisplayed etc. 

### How to reproduce the bug.

tsconfig file

```
{
  "compilerOptions": {
    "outDir": "./.tsbuild/",
    "target": "es2016",
    "module": "commonjs",
    "types": ["node", "webdriverio/sync", "@wdio/jasmine-framework"],
    "esModuleInterop": true,
    "allowSyntheticDefaultImports": true
  },
  "include": [
    "./appium/tests/**/*.ts"
  ]
}
```

package.json

```
"devDependencies": {
    "@babel/cli": "^7.20.7",
    "@babel/core": "^7.20.12",
    "@babel/eslint-parser": "^7.19.1",
    "@babel/preset-env": "^7.20.2",
    "@babel/register": "^7.18.9",
    "@babel/traverse": "^7.20.12",
    "@babel/types": "^7.20.7",
    "@ffmpeg-installer/ffmpeg": "^1.1.0",
    "@wdio/cli": "7.29.1",
    "@wdio/jasmine-framework": "7.26",
    "@wdio/local-runner": "7.29.1",
    "@wdio/spec-reporter": "7.29.1",
    "@wdio/types": "7.26.0",
    "@wdio/sync": "7.27",
    "axios": "0.27.2",
    "dotenv": "^16.0.1",
    "downloads-folder": "^3.0.3",
    "eslint": "^8.31.0",
    "eslint-config-standard": "^17.0.0",
    "eslint-plugin-import": "^2.26.0",
    "eslint-plugin-node": "^11.1.0",
    "eslint-plugin-promise": "^6.1.1",
    "eslint-plugin-wdio": "^7.25.3",
    "exceljs": "^4.3.0",
    "faker": "5.5.3",
    "file-type": "^16.0.1",
    "form-data": "^4.0.0",
    "jsdom": "^19.0.0",
    "moment": "^2.29.3",
    "node-fetch": "^2.6.1",
    "nodemailer": "^6.7.5",
    "pkce-challenge": "^3.0.0",
    "request": "^2.88.2",
    "testcafe": "^2.1.0",
    "testcafe-reporter-allure-expanded": "^1.1.0",
    "testcafe-reporter-teamcity": "^1.0.12",
    "testcafe-reporter-xunit": "^2.2.1",
    "ts-node": "^10.9.1",
    "typescript": "^4.9.4",
    "uuid": "^8.3.2",
    "wdio-teamcity-reporter": "^3.1.0",
    "webdriverio": "7.29.1",
    "xlsx": "^0.18.5"
  }
```

babel.config.js

`module.exports = {
    presets: [
        [ '@babel/preset-env', {
            targets: {
                node: 'current',
            },
        } ],
    ]
};
`

### Environment

- Operating system: MAC M1 Pro on Ventura 13.1 OSX
  - Appium CLI version (output of `appium --version`): 2.0.0-beta.48
  - Node.js version (output of `node --version`): v14.21.2
  - `npm` version (output of `npm --version`):  9.2.0
  - Last `appium` version which did _not_ exhibit the problem: 1.22.3
- Appium driver(s) and their version(s): xcuitest@4.16.6 (automationName 'XCUITest')
- Appium plugin(s) and their version(s): no plugins installed
- Platform and version under test: IOS 16.2
- Real device or emulator/simulator: simulator iphone 14 Pro

### Relevant log output

[AppiumLogs.zip](https://github.com/webdriverio/webdriverio/files/10401002/AppiumLogs.zip)

BUG01 attached example fail from spec reporter:
`[iPhone 14 Pro iOS 16.2 #0-0] 1) TestingJasmineWebdriverIoAsserts toBePresentNotWorking
[iPhone 14 Pro iOS 16.2 #0-0] Expected Element({ sessionId: '1de9329c-6a39-4fc8-b755-8bf3ff31e597', elementId: '0C000000-0000-0000-5A1C-000000000000', element-6066-11e4-a52e-4f735466cecf: '0C000000-0000-0000-5A1C-000000000000', selector: '~Get started', parent: Browser({ sessionId: '1de9329c-6a39-4fc8-b755-8bf3ff31e597', capabilities: Object({ webStorageEnabled: false, locationContextEnabled: false, browserName: '', platform: 'MAC', javascriptEnabled: true, databaseEnabled: false, takesScreenshot: true, networkConnectionEnabled: false, platformName: 'iOS', iosInstallPause: 5000, deviceName: 'iPhone 14 Pro', language: 'en', platformVersion: '16.2', automationName: 'XCUITest', app: '/Users/gfor/Documents/AutomationProject/TestAutomation/appium/apps/Repsly.app', newCommandTimeout: 10000, wdaLaunchTimeout: 300000, wdaConnectionTimeout: 600000, wdaStartupRetries: 4, wdaStartupRetryInterval: 120000, wdaLocalPort: 8668, udid: '96E22F2B-5DA1-4C51-973F-08E38416AA22' }), addCommand: Function, overwriteCommand: Function, addLocato ... to be present.      
      actual expected`

```typescript
I'm hoping to get some answers based on the provided info, I can't provide much because of my company privacy policy. But, if you can point me in the right direction, maybe I can get some stuff.
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [X] I have searched the existing issues

**Comment by gforepsly** [comment]
I will copy my answer from the other topic, maybe it helps someone as well:

"So, I did a lot of tries and I can say that the "sync" option in WebdriverIO is the cause of the issues. As soon as I installed everything from scratch and started fully with async (also preferred) WebdriverIO way - all of "expect" assertions are working as expected.

The problem is that our project has more than 2 years, so with every WebdriverIO major version update something new was different, which ultimately now means that even with Webdriver 7.x and NodeJs 14 (last versions where sync syntax should work) it doesn't work all they way. We are refactoring our code and that will solve all of our isses.

Thx for the tips - We also check for presence before some actions like click etc.; this example was just for the sake of example."
